Update hello tutorial - Closes #700 (#702)
LiskHQ/lisk-docs: modules/ROOT/pages/tutorials/hello-world.adoc

TIP: See the file on GitHub: {url_github_hello_index}[hello_world/index.js^].

<1> Require `Application` class, the devnet genesis block and the devnet config for the application.
The dependencies are required from the `lisk-sdk` package.

> The most important one is the `Application` class, which is used in <5> to create the `Application` instance.

Reviewer: ```suggestion
The most important dependency is the `Application` class, which is used in <5> to create the `Application` instance.
```

The `Application` instance will start the whole application at the bottom of `index.js`.
<2> Set the name of the blockchain application.

> <3> In the case whereby a different user other than `lisk` was given for access to the database `lisk_dev`, it will be necessary to update the username in the config.

Reviewer: ```suggestion
<3> In the case whereby a different user other than `lisk` was provided to gain access to the database `lisk_dev`, it will be necessary to update the username in the config.
```

[[step3]]
== 3. Create a new transaction type

For the Hello World App, it is necessary to create a xref:{url_customize}[custom transaction] `HelloTransaction`: +

If the account contains an adequate enough balance to process the `HelloTransaction` transaction, the new "hello" property will appear into the account’s asset field.

****
After sending a `HelloTransaction`, for example `{"type": 20, "senderId": "16313739661670634666L", ... "asset": { "hello": "world" } }`, the sender’s account will change from: +
`{ address: "16313739661670634666L", ..., asset: null }`, to +
`{ "address": "16313739661670634666L", ..., "asset": {"hello": "world"} }`.
****

> Now let's define the new transaction type, `HelloTransaction`.

Reviewer: ```suggestion
The next step is to define the new transaction type, `HelloTransaction` as described below.
```

TIP: Please see the file on Github: {url_github_hello_index}[hello_world/index.js^].

<1> *New line*: Require the newly created transaction type 'HelloTransaction'.
<2> *New line*: Register the 'HelloTransaction'.

After the 2 new lines shown above are added to your `index.js` file, save and close it.

== 5. Start the network

It should now be possible to start the customized blockchain network for the first time.

The parameter `configDevnet`, which is passed to the `Application` instance in <<step3,step 3>>, is preconfigured to start the node with a set of genesis delegates, that have enabled forging by default in the Devnet.

> These genesis delegates stabilize the new network, and ensure it is possible to test out the basic functionality of the network immediately with only one node, which is beneficial during development of the blockchain application.

Reviewer: ```suggestion
These genesis delegates stabilize the new network, and ensure it is possible to test out the basic functionality of the network immediately with only one node, which in turn is beneficial during development of the blockchain application.
```

[NOTE]
====
The genesis delegates can be replaced with real delegates later.

> For this, users needs to create new accounts and register themselves as delegates on the network.

Reviewer: ```suggestion
To enable this, users needs to create new accounts and register themselves as delegates on the network.
```

More information about this can be found in the guide how to xref:{url_guide_launch}[launch an application].
====

To start the network again, execute the following command shown below:

[source,bash]
----
node index.js
----

Please check the logs in order to to verify that the network has started successfully.

If any problems occur, then the process should stop and an error with debug information will be displayed.

== 6. Interact with the network

> Now with the network is running, try to send a `HelloTransaction` to the node to see if it will be accepted.

Reviewer: ```suggestion
Now with the network running, try to send a `HelloTransaction` to the node to see if it will be accepted.
```

    "signatures": [
        "db6cf96756fe537eebe427f40f8832d14a408e38ab2b0a40a8e584bf22642c59556dcf9452b9021205338ffc6db6e29dd3fc34795e66be26c6e88376382e6c02"
    ],
    "type": 20
}
----

This transaction object can be posted to a node.

More information about how to interact with the blockchain application can be found in the guide xref:{url_guide_interact}[Interact with the API].

=== A simple react frontend

Now that we know how to <<sendable,create sendable transaction objects>>, let's discover the application through a simple react frontend.

Please download the {url_github_hello_client}[react-client] folder and add it to the root of the Hello World application.

.Download the react-client folder, copy it into your project and install the dependencies
[source,bash]
----
cd ..
git clone git@github.com:LiskHQ/lisk-sdk-examples.git
cd hello_world
cp -r ../lisk-sdk-examples/hello_world/react-client react-client
cd react-client
npm i
----

.Starts the react client
[source,bash]
----
npm start
----

The Hello World frontend is now available under `localhost:8080`

IMPORTANT: To test the frontend, always ensure your node application is still running in the other terminal window.

> The react client offers a collection of simple scripts for the most common interactions for a blockchain application, and gives you a good start to extend and adjust the application further to fit your particular use case.

Reviewer: ```suggestion
The react client offers a collection of simple scripts for the most common interactions for a blockchain application. In addition, it provides you with a good start to extend and adjust the application further to fit your particular use case.
```

TIP: Check out the xref:{url_guide_frontend}[Connect a frontend] guide for a detailed description of the react client and it's different pages.

=== Optional: Run the app in the background

> For further interaction with the network, it is possible to run the process of the node in the background by executing the following commands:

Reviewer: ```suggestion
For further interaction with the network, it is possible to run the node in the background by executing the following commands:
```

    });
----

<1> Require `Application` class, the default genesis block and the default config for the application.
<2> Require the newly created transaction type `HelloTransaction`.
<3> Set the name of your blockchain application.

> <4> In the case whereby a different user than `lisk` was given, to access to the database `lisk_dev`, it is necessary to update the username in the config.

Reviewer: ```suggestion
<4> In the case whereby a different user than `lisk` was provided, to gain access to the database `lisk_dev`, it is necessary to update the username in the config.
```
